Remove a dead code from mod_proxy_cluster

jajik opened the pull request:
I looked into #63, trying to remove the dead code. It seems that the whole function can be removed without affecting the functionality. Tests in the repository are passing. Also, the described behaviour in JIRA related to the fix that introduced the bug is still not reproducible with this change – the node gets flagged REMOVED with hung JVM (the actual removal then depends on `MaxConnectionsPerChild`).

Please, check that I am not missing something.

jfclere reviewed (commented):
I don't get why you removed the code...

jajik commented:
> I don't get why you removed the code...

The logic was not in use for two years, so I assumed the current state is the one. Sorry about that.

----------------------

Anyway, I've dropped the previously proposed commit and removed the else branch as you suggested. Tests are passing and CI too, so I'll merge it now.
